#!/usr/bin/python2

#########################################################################
### Configuration #######################################################
#########################################################################

config_file = '/etc/ldapgroup/config.yaml'
description = 'list users in a specified LDAP group'
group_name = None

#########################################################################
### Declarations ########################################################
#########################################################################

import argparse
import ldapgroup
import sys

#########################################################################
### main () #############################################################
#########################################################################

parser = argparse.ArgumentParser(description=description)
parser.add_argument('group_name', type=str, help="group name")
parser.add_argument('--config', dest='config', type=str,
    default=config_file, help="alternate config file")

args = parser.parse_args()

config = ldapgroup.config_yaml(args.config)

group_name = args.group_name
try:
    connection = ldapgroup.connect(group_name=group_name, config=config)
except Exception as e:
    print("failed to connect as %s: %s" % (group_name, e))
    sys.exit(-1)

for user in ldapgroup.users(connection):
    print(user)
